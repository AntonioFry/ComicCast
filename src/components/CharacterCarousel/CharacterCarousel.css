.carousel-section {
  align-items: center;
  /* border: 1px solid black; */
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(5, 1fr);
  height: 300px;
  margin: 0px 0px 50px;
  overflow: hidden;
  overflow-x: scroll;
  overflow-y: hidden;
  padding: 10px;
}

.left {
    -ms-overflow-style: none;
    overflow: -moz-scrollbars-none;
}

::-webkit-scrollbar { 
    display: none; 
}