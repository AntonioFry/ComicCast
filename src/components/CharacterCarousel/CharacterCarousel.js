import React, { Component } from 'react'
import Card from '../Card/Card';
import './CharacterCarousel.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class CharacterCarousel extends Component {
  constructor(props) {
    super(props);
    this.state = {
      randomCharacters: []
    }
  }

  componentDidMount() {
    this.getRandomCharacters();
  }

  getRandomCharacters = () => {
    let allCharacters = this.props.allCharacters.slice();
    let randomCharacters = [];
  
    for (let i = 0; i < 5; i++) {
      const randomNumber = Math.floor(Math.random() * allCharacters.length - 1);
      const character = allCharacters.splice(randomNumber, 1);
      randomCharacters.push(...character)
    }
    this.setState({ randomCharacters })
  }
  
  render() {
    const characters = this.state.randomCharacters.map(character => {
      return (
        <Card
          name={character.name}
          id={character.id}
          image={character.image}
          key={character.id}
        />
      )
    })
    return (
      <section className="carousel-section">
        {characters}
      </section>
    )
  }
}

export const mapStateToProps = (store) => ({
  allCharacters: store.allCharacters
})

CharacterCarousel.propTypes = {
  allCharacters: PropTypes.array.isRequired
}

export default connect(mapStateToProps)(CharacterCarousel);