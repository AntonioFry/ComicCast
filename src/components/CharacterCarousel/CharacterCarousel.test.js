import React from 'react';
import { CharacterCarousel } from './CharacterCarousel';
import { shallow } from 'enzyme';
import { mapStateToProps } from './CharacterCarousel';

describe('CharacterCarousel', () => {

  describe('CharacterCarousel component', () => {
    let wrapper;
    let props;
  
    beforeEach(() => {
      props = {
        allCharacters: [{}, {}, {}, {}, {}]
      }
      wrapper = shallow(<CharacterCarousel allCharacters={props.allCharacters}/>)
    });
  
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should invoke getRandomCharcaters when componentDidMount is called', () => {
      wrapper.instance().getRandomCharacters = jest.fn()
      wrapper.instance().componentDidMount()
      expect(wrapper.instance().getRandomCharacters).toHaveBeenCalled();
    });

    it('should change state when getRandomCharacters', () => {
      wrapper.instance().getRandomCharacters();
      expect(wrapper.state('randomCharacters').length).toEqual(5)
    });
  })

  describe('mapStateToProps', () => {
    
    it('should return an object with all characters', () => {
      const mockStore = {
        allCharacters: [{}, {}, {}, {}, {}],
      }

      const expected = {
        allCharacters: [{}, {}, {}, {}, {}],
      }

      const mappedProps = mapStateToProps(mockStore);

      expect(mappedProps).toEqual(expected);
    });
  });
});