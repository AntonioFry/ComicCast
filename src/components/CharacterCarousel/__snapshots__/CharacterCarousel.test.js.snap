// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCarousel CharacterCarousel component should match the snapshot 1`] = `
<section
  className="carousel-section"
>
  <Card />
  <Card />
  <Card />
  <Card />
  <Card />
</section>
`;

exports[`CharacterCarousel should match the snapshot 1`] = `
<section
  className="carousel-section"
>
  <Card />
  <Card />
  <Card />
  <Card />
  <Card />
</section>
`;
